src/json/from_json.cr

```diff
@@ -73,16 +73,7 @@ def Bool.new(pull : JSON::PullParser)
   pull.read_bool
 end
 
-{% for type, method in {
-                         "Int8"   => "i8",
-                         "Int16"  => "i16",
-                         "Int32"  => "i32",
-                         "Int64"  => "i64",
-                         "UInt8"  => "u8",
-                         "UInt16" => "u16",
-                         "UInt32" => "u32",
-                         "UInt64" => "u64",
-                       } %}
+{% for type in %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) %}
   def {{type.id}}.new(pull : JSON::PullParser)
```

Comment:
Tried, not working. Got `Error: undefined method 'to_i64' for JSON::PullParser`.
With `--error-trace`, I can get this error too `Error: instantiating 'Int64.class#new(JSON::PullParser)'`.

It can be the result of overload priorities, `Int64.new(value)` being taken first over `Int64.new(pull : JSON::PullParser)`.

Reply:
Note there is no such issue with YAML, since two arguments are taken.